### PR TITLE
EVA-3649 Find studies for import accession

### PR DESCRIPTION
**Pull request by nitin-ebi**
As part of this PR only [tasks/eva_3649/findStudiesForImportAccession.py](https://github.com/EBIvariation/eva-tasks/compare/master...nitin-ebi:eva-tasks:find_studies_for_import_accession?expand=1#diff-8c5fe2ccf98e0cb39c97438df57145f5733253b138a59c975df7bb667ae74ea1) needs to be reviewed.
Other changes are from [EVA2038](https://www.ebi.ac.uk/panda/jira/browse/EVA-2038) for which there is already a separate PR waiting to be reviewed and merged

### CODE REVIEW

File: tasks/eva_3649/findStudiesForImportAccession.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                    collection = database["variants_2_0"]
+
+                    pipeline = [
+                        {"$match": {"ids": {"$not": {"$regex": "^ss.+$"}}}},
+                        {"$unwind": "$files"},
+                        {"$group": {"_id": db_name, "sids": {"$addToSet": "$files.sid"}}}
+                    ]
+                    result = collection.aggregate(pipeline, allowDiskUse=True)
+
+                    cursor = result
```
Comment (apriltuesday) [inline tasks/eva_3649/findStudiesForImportAccession.py:67]:
```suggestion
                    cursor = collection.aggregate(pipeline, allowDiskUse=True)

```

File: tasks/eva_3649/findStudiesForImportAccession.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    if cursor:
+                        logger.info(f"Studies found in database: {db_name}")
+                        for doc in cursor:
+                            studies_list = doc.get("sids", [])
+                            logger.info(f"Studies List: {studies_list}")
+                            try:
+                                with open(db_file_path, 'w') as file:
+                                    for study in studies_list:
+                                        file.write(f"{study}\n")
+                                print("List has been written to the file successfully.")
```
Comment (apriltuesday) [inline tasks/eva_3649/findStudiesForImportAccession.py:77]:
```suggestion
                                logger.info("List has been written to the file successfully.")
```

File: tasks/eva_3649/findStudiesForImportAccession.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        logger.error(f"Error reading files: {e}")
+
+    logger.info(f"Final Studies List: {final_studies_list}")
+
+    try:
+        final_file_path = os.path.join(study_dir, f"finalList_{datetime.now()}.txt")
+        with open(final_file_path, 'w') as file:
+            for study in final_studies_list:
+                file.write(f"{study}\n")
+        print("List has been written to the file successfully.")
```
Comment (apriltuesday) [inline tasks/eva_3649/findStudiesForImportAccession.py:35]:
```suggestion
        logger.info("List has been written to the file successfully.")
```

**Review by apriltuesday — APPROVED** [review]

File: tasks/eva_3649/findStudiesForImportAccession.py (hunk 3 above)
Comment (nitin-ebi) [reply tasks/eva_3649/findStudiesForImportAccession.py]: done

File: tasks/eva_3649/findStudiesForImportAccession.py (hunk 2 above)
Comment (nitin-ebi) [reply tasks/eva_3649/findStudiesForImportAccession.py]: done

**Pull request merged**